sdks/python/apache_beam/transforms/window_test.py

```diff
@@ -252,6 +276,50 @@ def test_timestamped_with_combiners(self):
       assert_that(mean_per_window, equal_to([(0, 2.0), (1, 7.0)]),
                   label='assert:mean')
 
+  @attr('ValidatesRunner')
+  def test_custom_windows(self):
+    with TestPipeline() as p:
+      result = (p | Create([TimestampedValue(x, x * 100) for x in range(7)])
+                | 'custom window' >> WindowInto(TestCustomWindows(250, 200))
+                | 'insert key' >> Map(lambda v: ('key', v.value))
+                | GroupByKey())
+
+      assert_that(result, equal_to([('key', [0, 1, 2]),
+                                    ('key', [3, 4]),
+                                    ('key', [5]),
+                                    ('key', [6])]))
+
+  @attr('ValidatesRunner')
+  def test_windows_idempotency(self):
+    with TestPipeline() as p:
+      result = (p | Create([(x, x * 2) for x in range(5)])
+                | Map(lambda item: TimestampedValue(item[0], item[1]))
+                | Map(lambda v: ('key', v))
+                | 'window' >> WindowInto(FixedWindows(4))
+                | 'same window' >> WindowInto(FixedWindows(4))
+                | 'same window again' >> WindowInto(FixedWindows(4))
+                | GroupByKey())
+
+      assert_that(result, equal_to([('key', [0, 1]),
+                                    ('key', [2, 3]),
+                                    ('key', [4])]))
+
+  @attr('ValidatesRunner')
+  def test_windows_gbk_idempotency(self):
+    with TestPipeline() as p:
+      result = (p | Create([(x, x * 2) for x in range(5)])
```

Comment:
Ditto for timestamped values being done in Create.

Reply:
Same for this one:)